tools/xskperfsuite.ps1

```diff
@@ -213,7 +216,12 @@ try {
                             Write-Host $($Format -f $ScenarioName, [Math]::ceiling($avg), [Math]::ceiling($stddev))
 
                             if (-not [string]::IsNullOrEmpty($RawResultsFile)) {
-                                Add-Content -Path $RawResultsFile -Value ("{0},{1}" -f $ScenarioName, ($kppsList -join ","))
+                                Add-Content -Path $RawResultsFile -Value `
+                                    ("{0},{1},{2},{3}" -f `
+                                        $ScenarioName, `
+                                        $CommitHash, `
+                                        ([DateTimeOffset](Get-Date)).ToUnixTimeSeconds(), `
```

Comment:
Have you considered using the time of commit as opposed to the time of test?

(I'm not sure whether this is a good idea, just that it's an interesting idea, and I know you can derive the time of commit from the commit hash separately but it could be a real convenience)

(maybe I'm overthinking things here)

Reply:
If the suggestion here is that the time of commit is also useful, I agree. Looking that up from source control is probably best until proven inadequate (a single source of truth is best) and we could look that up and retroactively update our data files if we change the schema.

Reply:
I suppose a practical dashboard will want to render the data onto a time scale, though, and in that case, is the time of commit the most relevant time? It probably is. But is the most important relationship that one of parent commit and child commit, independent of the number of days between commits?

Reply:
Now I'm thinking a real database might actually be a good solution, because it makes sorting by commit/time and filtering by branches easy.

Reply:
(i.e. shasum A comes after shasum B in the csv if and only if A comes after B in the git graph)

Reply:
It would definitely be problematic for this solution, which depends on Azure appending atomically to the end of the data file. We'd need something providing transactions (hmm, git?) if we want to modify the data file anywhere else.

Reply:
Yeah, makes sense. I was basically asking whether we will ever test a commit that isn't the latest, but that would be a pretty restrictive assumption.

Reply:
I wouldn't worry too much about this right now. As you improve what you collect and store, my expectation is that you won't be able to store everything in just one simple file. For instance, per-run latency graphs, are something that have a ton of data points per run, and can't be stored in one file easily with everything else (IMO).

As a result, you inevitably have a model where you have all the raw data, and then usually some post processed version just for the dashboard (though you _may_ be able to do this processing dynamically, but at a cost to download more files and process more data).